fendermint/app/src/cmd/run.rs

```diff
@@ -50,7 +61,26 @@ async fn run(settings: Settings) -> anyhow::Result<()> {
         NamespaceBlockstore::new(db.clone(), ns.state_store).context("error creating state DB")?;
 
     let resolve_pool = CheckpointPool::new();
-    let parent_finality = Arc::new(create_parent_finality(&settings).await?);
+    let multi_engine = Arc::new(MultiEngine::new(1));
+    let db = Arc::new(db);
+    let state_store = Arc::new(state_store);
+
+    // setup top down parent finality related code
+    let parent_finality_getter = ParentFinalityQuery::new(
+        db.clone(),
+        state_store.clone(),
+        multi_engine.clone(),
+        ns.app.clone(),
+    );
+    let parent_finality = Arc::new(create_parent_finality(&settings, &parent_finality_getter)?);
+    let ipc_agent_proxy = create_ipc_agent_proxy(&settings)?;
+    let polling_parent_syncer = PollingParentSyncer::new(
+        settings.parent_finality.clone(),
+        parent_finality.clone(),
+        Arc::new(ipc_agent_proxy),
+    );
+    polling_parent_syncer.start()?;
```

Comment:
Can you remind me how are errors handled by this routine? 

Also, we should be careful once we introduce the ability to sync a node from scratch, as we may start fetching parent finality that is outdated while our peer is still syncing with the chain.

Reply:
For now, it would just catch the error, and retry execution.